### PR TITLE
V1.4.0+galaxy0

### PR DESCRIPTION
**Pull request by RJMW**

### CODE REVIEW

**Comment by RJMW** [comment]
@simonbray Please, can you review this PR.

File: tools/dimspy/process_scans.xml [hunk 1]
```diff
@@ -6,27 +6,25 @@
...
-                #if str( $fn ).endswith(".dat")
+                #if str($fn).endswith('.dat')
                     ln -s '$fn' '$fn.name'
             	    &&
                 #end if
             #end for
-        #elif $input.format == "single_file"
-            #if str( $input.source ).endswith(".dat")
+        #elif $input.format == 'single_file'
+            #if str($input.source).endswith('.dat')
```
Comment (bgruening) [inline tools/dimspy/process_scans.xml]: It always is .dat, what are you trying to do here?

File: tools/dimspy/process_scans.xml [hunk 2]
```diff
@@ -6,27 +6,25 @@
...
-            #for $fn in $input.source
-                --input '$fn.name'
-            #end for
-        #elif $input.format == "single_file"
+        #if $input.format == 'data_collection'
+            --input .
+        #elif $input.format == 'single_file'
             --input '$input.source.name'
-        #elif $input.format == "library"
+        #elif $input.format == 'library'
```
Comment (bgruening) [inline tools/dimspy/process_scans.xml]: remove this option

File: tools/dimspy/process_scans.xml [hunk 3]
```diff
@@ -195,6 +193,31 @@
                 <element name="batch04_QC18_rep03_282" file="batch04_QC18_rep03_282.txt" ftype="tsv"/>
             </output_collection>
         </test>
+        <test>
+            <param name="input|format" value="data_collection" />
```
Comment (bgruening) [inline tools/dimspy/process_scans.xml]: I think the better syntax is this here:

https://github.com/galaxyproject/tools-iuc/blob/57493fdc88e6128bed127993464f2a6480e2774d/tools/enasearch/enasearch_search_data.xml#L82

**Review by bgruening — COMMENTED** [review]
* [x] size property can be removed
* [x] zip file should be also removed ideally
* [x] remove the library option
* [ ] `#if float($adv.ringing_threshold) > 0.0` --> the casting to float should not be necessary 
* [x] merge data_collection with single_file by using `multiple="True"`
* [x] remove sim_size as much as you can
  * [ ] use asserts for hdf5
  * [x] and normal diff for text files

**Comment by Tomnl** [comment]
* [x] Perhaps replace all "name" attributes to using "argument". The name will be automatically be the argument (without the hyphens).e.g
   * Replace ` <param name="snr_threshold" type="float" value="3.0" label="Signal-to-noise ratio threshold" help="" argument="--snr-threshold" />`
   * With  ` <param argument="--snr-threshold" type="float" value="3.0" label="Signal-to-noise ratio threshold" help=""  />`
   * Note that the reference in the cheetah code might need to be checked
* [x] Stick to recommended ordering of arguments in param where possible
    * name
    * argument
    * type
    * format
    * min | truevalue
    * max | falsevalue
    * value | checked
    * optional
    * label
    * help
*  [x] Text validators - are there any situations where certain text should be avoided for some params e.g. https://github.com/computational-metabolomics/dimspy-galaxy/blob/v1.4.0%2Bgalaxy0/tools/dimspy/sample_filter.xml#L35

**Comment by bgruening** [comment]
@RJMW why not moving this into the other repo where you already have the travis setup?

**Review by Tomnl — APPROVED** [review]

**Review by bgruening — COMMENTED** [review]
Cool! Happy Easter!

**Comment by RJMW** [comment]
@bgruening Happy Easter to you, too!

**Pull request merged**